From 33a9ec1309cf9e4c8b64f289acf509c81d5bb3f8 Mon Sep 17 00:00:00 2001
From: mshawcroft <mshawcroft@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Mon, 20 Jan 2014 17:41:54 +0000
Subject: [PATCH] [AArch64] Define BE loader name.

	Backport from mainline
	2014-03-19  Victor Kamensky  <victor.kamensky@linaro.org>

	    2014-01-20  Marcus Shawcroft  <marcus.shawcroft@arm.com>

	    * config/aarch64/aarch64-linux.h (GLIBC_DYNAMIC_LINKER): Expand
	    loader name using mbig-endian.
	    (LINUX_TARGET_LINK_SPEC): Pass linker -m flag.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@206845 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/config/aarch64/aarch64-linux.h | 5 +++--
 2 files changed, 9 insertions(+), 2 deletions(-)

Index: gcc-linaro-4.8-2014.03/gcc/config/aarch64/aarch64-linux.h
===================================================================
--- gcc-linaro-4.8-2014.03.orig/gcc/config/aarch64/aarch64-linux.h
+++ gcc-linaro-4.8-2014.03/gcc/config/aarch64/aarch64-linux.h
@@ -21,7 +21,7 @@
 #ifndef GCC_AARCH64_LINUX_H
 #define GCC_AARCH64_LINUX_H
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64.so.1"
+#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-aarch64%{mbig-endian:_be}.so.1"
 
 #define CPP_SPEC "%{pthread:-D_REENTRANT}"
 
@@ -32,7 +32,8 @@
    %{rdynamic:-export-dynamic}			\
    -dynamic-linker " GNU_USER_DYNAMIC_LINKER "	\
    -X						\
-   %{mbig-endian:-EB} %{mlittle-endian:-EL}"
+   %{mbig-endian:-EB} %{mlittle-endian:-EL}     \
+   -maarch64linux%{mbig-endian:b}"
 
 #define LINK_SPEC LINUX_TARGET_LINK_SPEC
 
